### ISSUE TITLE
Nested build stages don't nest properly in console 

### ISSUE THREAD

**Issue by timja** [jenkinsci/pipeline-graph-view-plugin#92]
### Jenkins and plugins versions report

-

### What Operating System are you using (both controller, and any agents involved in the problem)?

-

### Reproduction steps

see
https://ci.jenkins.io/job/Plugins/job/slack-plugin/job/master/325/pipeline-console/?selected-node=25

it uses https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy

I'll get a minimal pipeline up later on that replicates the same thing

The steps should be nested under the build stage but instead they are nested under linux-11

![image](https://user-images.githubusercontent.com/21194782/190622285-8e9ee0bd-9afd-417f-b0f8-9d515f19e116.png)

### Expected Results

steps nested under build

### Actual Results

steps nested under linux-11

**Comment by timbrown5** [comment]
I've tried nested Declarative Pipeline stages and they seem to work:

![Screenshot 2022-09-30 at 11 27 27](https://user-images.githubusercontent.com/4447764/193251031-141b8d77-9e6a-4818-aba5-509839a2af07.png)

```
pipeline {
    agent any

    stages {
        stage('Non-Parallel Stage') {
            steps {
                echo 'This stage will be executed first.'
                // Test empty/no args for step
                echo ''
            }
        }
        stage('Parallel Stage') {
            failFast true
            parallel {
                stage('Branch A') {
                    steps {
                        echo "On Branch A - 1"
                        echo "On Branch A - 2"
                    }
                }
                stage('Branch B') {
                    steps {
                        echo "On Branch B - 1"
                        echo "On Branch B - 2"
                    }
                }
                stage('Branch C') {
                    stages {
                        stage('Nested 1') {
                            stages {
                                stage('Nested 1.1') {
                                    stages {
                                        stage('Nested 1.1.1') {
                                            stages {
                                                stage('Nested 1.1.1.1') {
                                                    stages {
                                                        stage('Nested 1.1.1.1.1') {
                                                            stages {
                                                                stage('Nested 1.1.1.1.1.1') {
                                                                    steps {
                                                                        echo "In stage Nested 1.1.1.1.1.1 - 1 within Branch C"
                                                                        echo "In stage Nested 1.1.1.1.1.1 - 2 within Branch C"
                                                                    }
                                                                }
                                                                stage('Nested 1.1.1.1.1.2') {
                                                                    steps {
                                                                        echo "In stage Nested 1.1.1.1.1.2 - 1 within Branch C"
                                                                        echo "In stage Nested 1.1.1.1.1.2 - 2 within Branch C"
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
        stage('Skipped stage') {
            when {
                branch 'master'
            }
            steps {
                echo 'This stage will be skipped.'
            }
        }
    }
}
```

**Comment by timbrown5** [comment]
Works with nested scripted too :/.

![Screenshot 2022-09-30 at 11 37 15](https://user-images.githubusercontent.com/4447764/193252540-85416459-95d5-43cf-a3be-909ddef8ee59.png)


```
node {
    stage('Non-Parallel Stage') {
        echo 'This stage will be executed first.'
        // Test empty/no args for step
        echo ''
    }
    stage('Parallel Stage') {
        parallel 'Branch A': {
            echo "On Branch A - 1"
            echo "On Branch A - 2"
        },
        'Branch B': {
            echo "On Branch B - 1"
            echo "On Branch B - 2"
        },
        'Branch C': {
            stage('Nested 1') {
                stage('Nested 1.1') {
                    stage('Nested 1.1.1') {
                        stage('Nested 1.1.1.1') {
                            stage('Nested 1.1.1.1.1') {
                                stage('Nested 1.1.1.1.1.1') {
                                    echo "In stage Nested 1.1.1.1.1.1 - 1 within Branch C"
                                    echo "In stage Nested 1.1.1.1.1.1 - 2 within Branch C"
                                }
                                stage('Nested 1.1.1.1.1.2') {
                                    echo "In stage Nested 1.1.1.1.1.2 - 1 within Branch C"
                                    echo "In stage Nested 1.1.1.1.1.2 - 2 within Branch C"
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```
I don't think I have access to view the linked pipeline to see what is different/

**Comment by timbrown5** [comment]
I think this is a repo:

![Screenshot 2022-09-30 at 12 14 05](https://user-images.githubusercontent.com/4447764/193258024-61b5891d-53c3-4a86-b2af-538558261583.png)

```
node {
    stage("Build all") {
    	parallel "linux-8": {
    		retry(3) {
				timeout(time: 60, unit: 'SECONDS') {
					stage ("Checkout (linux-8)") {
						echo "Checking out linux-8"
					}
					stage ("Build (linux-8)") {
						echo "Building linux-8"
					}
					stage ("Archive (linux-8)") {
						echo "Archiving linux-8"
					}
				}
    	    }
    	},
    	"linux-11": {
    		retry(3) {
				timeout(time: 60, unit: 'SECONDS') {
					stage ("Checkout (linux-11)") {
						echo "Checking out linux-8"
					}
					stage ("Build (linux-11)") {
						echo "Building linux-8"
					}
					stage ("Archive (linux-11)") {
						echo "Archiving linux-8"
					}
				}
    		}
    	}
    }
	stage ("Deploy") {
		echo "Deploying..."
	}
}
```

**Comment by timbrown5** [comment]
I suspect it's another instance of differing expectations from the Graph and Tree view.

If I run the Pipeline below I get: 
```
node {
    stage("Build all") {
        stage ("Checkout (linux-8)") {
            echo "Checking out linux-8"
        }
        stage ("Build (linux-8)") {
            echo "Building linux-8"
        }
        stage ("Archive (linux-8)") {
            echo "Archiving linux-8"
        }
    }
}
```
This graph:
![Screenshot 2022-09-30 at 13 45 07](https://user-images.githubusercontent.com/4447764/193272609-57db733a-cf85-4dec-9ea8-cdb207efc748.png)

First question: is this expected (should the child stages be represented on the graph?).

![Screenshot 2022-09-30 at 13 45 01](https://user-images.githubusercontent.com/4447764/193272694-84ba8d79-ed86-49e1-8575-a255f3fb6f78.png)

This then causes the Console view to render incorrectly as there are steps returned from the API that has no parent stage (so don't get added to the graph).

- This might be slightly different to the original behaviour, but I think the causes are related).

**Comment by timbrown5** [comment]
> I'd expect the implicit "Check out from version control" step (which was executed at the start of the "Agent" stage) to be listed above the "Build" stage, but it is instead listed below the "Cleanup" stage. The steps in the nested stages are shown correctly.

This is likely fixed by: #136.
